NumLib/DOF/DOFTableUtil.cpp

```diff
@@ -100,24 +86,62 @@ double norm(GlobalVector const& x, unsigned const global_component,
             MathLib::VecNormType norm_type,
             LocalToGlobalIndexMap const& dof_table, MeshLib::Mesh const& mesh)
 {
-    switch(norm_type)
+    double res = 0.0;
+    MeshLib::MeshSubsets const& mss = dof_table.getMeshSubsets(global_component);
+
+    for (unsigned i=0; i<mss.size(); i++)
     {
-    case MathLib::VecNormType::NORM1:
-        return norm(
-            x, global_component, dof_table, mesh,
-            [](double res, double value) { return res + std::abs(value); });
-    case MathLib::VecNormType::NORM2:
-        return std::sqrt(
-            norm(x, global_component, dof_table, mesh,
-                 [](double res, double value) { return res + value * value; }));
-    case MathLib::VecNormType::INFINITY_N:
-        return norm(x, global_component, dof_table, mesh,
-                    [](double res, double value) {
-                        return std::max(res, std::abs(value));
-                    });
-    default:
-        OGS_FATAL("An invalid norm type has been passed.");
+        MeshLib::MeshSubset const& ms = mss.getMeshSubset(i);
+        if (ms.getMeshID() != mesh.getID())
+            continue;
+        switch (norm_type)
```

Comment:
IMHO you could have kept most of the old code.
Isn't it just necessary to add the `#ifdef PETSC` part (and remove the return statements from the original switch-case?